go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -301,18 +293,20 @@ func (oc *Controller) deleteEgressFirewall(egressFirewall *egressfirewallapi.Egr
 	klog.Infof("Deleting egress Firewall %s in namespace %s", egressFirewall.Name, egressFirewall.Namespace)
 	deleteDNS := false
 
-	nsInfo := oc.getNamespaceLocked(egressFirewall.Namespace)
-	if nsInfo != nil {
-		// clear it so an error does not prevent future egressFirewalls
-		for _, rule := range nsInfo.egressFirewall.egressRules {
+	oc.egressFirewallsMutex.Lock()
+	ef := oc.egressFirewalls[egressFirewall.Namespace]
+	if ef != nil {
+		for _, rule := range ef.egressRules {
 			if len(rule.to.dnsName) > 0 {
 				deleteDNS = true
 				break
 			}
 		}
-		nsInfo.egressFirewall = nil
-		nsInfo.Unlock()
 	}
+	// clear the egressFirewall so an error does not prevent future egressFirewalls
+	delete(oc.egressFirewalls, egressFirewall.Namespace)
+	oc.egressFirewallsMutex.Unlock()
+
```

Comment:
Not sure why we are holding `egressFirewallsMutex` for L298 to 305. Can this just be:

```
oc.egressFirewallsMutex.Lock()
ef := oc.egressFirewalls[egressFirewall.Namespace]
delete(oc.egressFirewalls, egressFirewall.Namespace)
oc.egressFirewallsMutex.Unlock()
if er != nil {
}
```

Reply:
I changed it to use a sync map I dont think this comment still applies